src/lib.rs

```diff
@@ -97,16 +97,56 @@
 
 mod collector;
 
+use std::sync::Arc;
+
 use metrics::{describe_gauge, gauge, Unit};
 
+/// Metrics names
+#[derive(Debug, PartialEq, Eq)]
+struct Metrics {
+    cpu_seconds_total: Arc<str>,
+    open_fds: Arc<str>,
+    max_fds: Arc<str>,
+    virtual_memory_bytes: Arc<str>,
+    virtual_memory_max_bytes: Arc<str>,
+    resident_memory_bytes: Arc<str>,
+    start_time_seconds: Arc<str>,
+    threads: Arc<str>,
+}
+
+impl Metrics {
+    // Create new Metrics, allocating prefixed strings for metrics names.
+    fn new(prefix: impl AsRef<str>) -> Self {
+        let prefix = prefix.as_ref();
+        Self {
+            cpu_seconds_total: format!("{prefix}process_cpu_seconds_total").into(),
+            open_fds: format!("{prefix}process_open_fds").into(),
+            max_fds: format!("{prefix}process_max_fds").into(),
+            virtual_memory_bytes: format!("{prefix}process_virtual_memory_bytes").into(),
+            virtual_memory_max_bytes: format!("{prefix}process_virtual_memory_max_bytes").into(),
+            resident_memory_bytes: format!("{prefix}process_resident_memory_bytes").into(),
+            start_time_seconds: format!("{prefix}process_start_time_seconds").into(),
+            threads: format!("{prefix}process_threads").into(),
+        }
+    }
+}
+
+impl Default for Metrics {
+    // Create new Metrics, without prefixing and thus allocating.
+    fn default() -> Self {
+        Self::new("")
+    }
+}
+
 /// Prometheus style process metrics collector
 #[derive(Debug, PartialEq, Eq, Clone)]
 pub struct Collector {
-    prefix: String,
+    metrics: Arc<Metrics>,
```

Comment:
Why not just `Metrics` here?

Reply:
To reduce the size of `Collector` from 128 bytes to 8 bytes. Also, `Clone`ing
`Collector`s would be much cheaper.